packages/components/src/lib/flamegraph.js

```diff
@@ -0,0 +1,66 @@
+const PRECISION = 10e6;
+
+// TODO aggregate with codeObject.fqid
+
+const digestEvent = (event) => ({
+  name: String(event.toString()),
+  value: Number(event.elapsedTime),
+  children: event.children.map(digestEvent),
+});
+
+const validateValue = (name, total, value) => {
```

Comment:
Negative time are definitely a bug. But I don't want to panic and just return 0. I left the explicit `if` instead of `Math.max` to be able to easily log the case were a function's children take more time that the functions itself which also should not happen.